freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java

```diff
@@ -32,26 +52,55 @@
     @Inject
     private FreeIpaClientFactory freeIpaClientFactory;
 
-    private boolean addService() {
-        return true;
-    }
-
-    private boolean addHost() {
-        return true;
-    }
-
     public ServiceKeytabResponse generateServiceKeytab(ServiceKeytabRequest request) {
+        LOGGER.debug("Request to generate keytab for Service:{} Host:{} in Environment:{}", request.getServiceName(), request.getServerHostName(),
+                request.getEnvironmentCrn());
         ServiceKeytabResponse response = new ServiceKeytabResponse();
-        // TODO add host and add pricipal before generating keytab.
-        //TODO  Will be implemented CDPSDX-515
-        response.setKeytab(null);
+        FreeIpaClient ipaClient;
+        Stack freeIpaStack = getFreeIpaStack(request.getEnvironmentCrn());
+        if (null == freeIpaStack) {
+            throw new RuntimeException("Stack for IPA Not Found");
+        }
+        String realm = getRealm(freeIpaStack);
+        ipaClient = getFreeIpaClient(freeIpaStack);
+        if (ipaClient == null) {
```

Comment:
this should never happen, as the creation will throw an exception if fails

Reply:
Will update the patch.